src/main/java/com/amashchenko/maven/plugin/gitflow/AbstractGitFlowMojo.java

```diff
@@ -316,10 +319,19 @@ private String getCurrentProjectOutputTimestamp() throws MojoFailureException {
      * @throws MojoFailureException
      *             If project loading fails.
      */
-    private MavenProject reloadProject(MavenProject project) throws MojoFailureException {
+    private MavenProject reloadProject(final MavenProject project) throws MojoFailureException {
         try {
-            ProjectBuildingResult result = projectBuilder.build(project.getFile(), mavenSession.getProjectBuildingRequest());
-            return result.getProject();
+            List<ProjectBuildingResult> result = projectBuilder.build(
+                    Collections.singletonList(project.getFile()),
+                    true,
+                    mavenSession.getProjectBuildingRequest());
+
+            for (ProjectBuildingResult projectBuildingResult : result) {
+                MavenProject resultProject = projectBuildingResult.getProject();
+                if (!resultProject.isExecutionRoot()) continue;
```

Comment:
Any reason to use `continue` instead of positive `if` and `return`? Also can you use brackets `{}` even for one line if.

Reply:
With JDK 8 onward I would write it like
```
    private MavenProject reloadProject(final MavenProject project) throws MojoFailureException {
        try {
            return projectBuilder
                    .build(
                            Collections.singletonList(project.getFile()),
                            true,
                            mavenSession.getProjectBuildingRequest())
                    .stream()
                    .map(ProjectBuildingResult::getProject)
                    .filter(MavenProject::isExecutionRoot)
                    .findAny().orElseThrow(() -> new NoSuchElementException(
                            "None of the  ProjectBuildingResults is for the executionRoot"));
        } catch (Exception e) {
            throw new MojoFailureException("Error re-loading project info", e);
        }
    }
```
The notation with continue seems closer to the stream filter notation to me. But I don't mind changing it to a positive if condition.